src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -146,6 +147,97 @@ pub trait MetadataChainExt: MetadataChain {
     fn iter_blocks_ref<'a>(&'a self, head: &'a BlockRef) -> DynMetadataStream<'a> {
         self.iter_blocks_interval_ref(head, None)
     }
+
+    async fn accept<E>(
```

Comment:
Plz add comments for new methods. It seems all methods in this extension are commented